src/taskgraph/transforms/fetch.py

```diff
@@ -9,91 +9,76 @@
 import os
 import re
 from dataclasses import dataclass
-from textwrap import dedent
-from typing import Callable
+from typing import Any, Callable, Dict, Optional
 
-from voluptuous import Extra, Optional, Required
+import msgspec
 
 import taskgraph
 
 from ..util import path
 from ..util.cached_tasks import add_optimization
-from ..util.schema import Schema, validate_schema
+from ..util.schema import Schema
 from ..util.treeherder import join_symbol
 from .base import TransformSequence
 
 CACHE_TYPE = "content.v1"
 
+
 #: Schema for fetch transforms
-FETCH_SCHEMA = Schema(
-    {
-        Required(
-            "name",
-            description=dedent(
-                """
-                Name of the task.
-                """.lstrip()
-            ),
-        ): str,
-        Optional(
-            "task-from",
-            description=dedent(
-                """
-                Relative path (from config.path) to the file the task was defined
-                in.
-                """.lstrip()
-            ),
-        ): str,
-        Required(
-            "description",
-            description=dedent(
-                """
-                Description of the task.
-                """.lstrip()
-            ),
-        ): str,
-        Optional("expires-after"): str,
-        Optional("docker-image"): object,
-        Optional(
-            "fetch-alias",
-            description=dedent(
-                """
-                An alias that can be used instead of the real fetch task name in
-                fetch stanzas for tasks.
-                """.lstrip()
-            ),
-        ): str,
-        Optional(
-            "artifact-prefix",
-            description=dedent(
-                """
-                The prefix of the taskcluster artifact being uploaded.
-                Defaults to `public/`; if it starts with something other than
-                `public/` the artifact will require scopes to access.
-                """.lstrip()
-            ),
-        ): str,
-        Optional("attributes"): {str: object},
-        Required("fetch"): {
-            Required("type"): str,
-            Extra: object,
-        },
-    }
-)
+class FetchConfig(Schema, rename=None, omit_defaults=False):
+    """Configuration for a fetch task type."""
+
+    type: str
+    # Additional fields handled dynamically by fetch builders
+
+
+class FetchSchema(Schema):
```

Comment:
I've noticed you sometimes use `FooSchema`, sometimes `FooConfig` and sometime neither. It would be good to be consistent across all schema class names.

Personally I think I like `FooSchema` the best (even for things that are sub schemas). Though happy to bikeshed with you if you have another preference.